packages/web3-core/package.json

```diff
@@ -16,7 +24,10 @@
 	"scripts": {
 		"clean": "rimraf dist && rimraf lib",
 		"prebuild": "yarn clean",
-		"build": "tsc --build",
+		"build": "yarn build:cjs && yarn build:esm && yarn build:types",
+		"build:cjs": "tsc --build tsconfig.cjs.json && echo '{\"type\": \"commonjs\"}' > ./lib/commonjs/package.json",
+		"build:esm": "tsc --build tsconfig.esm.json && echo '{\"type\": \"module\"}' > ./lib/esm/package.json",
```

Comment:
It was explicitly required to have package json in build dir ( lib ) to explicitly override module type for correctly loading ESM when required so I am generating here a package json in lib

Reply:
Woah, this looks so wrong. I don't think this is what you should be doing. There should be only one package.json at root of library.

You might be encountering typescript problem, I think each package.json needs to also have:
```
"main": "lib/cjs/index.js",
"module": "lib/esm/index.js",
```

Reply:
No idea tbh

Reply:
@jdevcs Maybe it would be better to use nodejs script for this. Afaik this excludes developers on windows to build library^^

Reply:
for creating package.json? I think `echo` is also avalible in win